### PR TITLE
HoC Interstitial Modal

### PR DESCRIPTION
**Pull request by AndrewJakubowicz**
- Add HoC interstitial modal.
- Repair vue destruction logic to properly destroy vue modals.
- Sessions storage used to show modal once per activity.

### CODE REVIEW

**Comment by AndrewJakubowicz** [comment]
SessionStorage is emptied when the browser is closed.

File: app/views/special_event/HoC2018InterstitialModal.vue [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    .modal-body
+      .row(v-if="showVideo")
+        .hoc-video
+          <iframe width="560" height="315" src="https://www.youtube.com/embed/SzZiOkZ-vRk" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
+      .row.buttons
+        .col-xs-6
+          button.btn.btn-lg.btn-navy(v-on:click="clickTeacher") I'm an educator
+          p Show me teacher resources!
+        .col-xs-6
+          button.btn.btn-lg.btn-navy(v-on:click="clickStudent") I'm a student
```
Comment (differentmatt) [inline app/views/special_event/HoC2018InterstitialModal.vue]: No text localization?

Comment (AndrewJakubowicz) [reply app/views/special_event/HoC2018InterstitialModal.vue]: Sorry. Great pickup.

**Comment by AndrewJakubowicz** [comment]
@differentmatt Would you be able to review changes. Thank you.

**Pull request merged**